src/main/java/org/broadinstitute/hellbender/utils/nio/PathLineIterator.java

```diff
@@ -0,0 +1,59 @@
+package org.broadinstitute.hellbender.utils.nio;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+/**
+ * Iterate through the lines of a Path. Works for anything you can point
+ * a Path to, such as a normal file or anything you have an NIO provider
+ * for (e.g. GCS).
+ */
+public class PathLineIterator implements AutoCloseable, Iterable<String> {
+  private final Stream<String> lines;
```

Comment:
Indentation looks weird in this file compare to others...

Reply:
Switched to 4-spaces indent